client/ayon_houdini/api/plugin.py

```diff
@@ -354,3 +355,35 @@ class HoudiniExtractorPlugin(publish.Extractor):
 
     hosts = ["houdini"]
     settings_category = SETTINGS_CATEGORY
+
+    def render_rop(self, instance: pyblish.api.Instance):
+        """Render the ROP node of the instance.
+
+        If `instance.data["frames_to_fix"]` is set and is not empty it will
+        be interpreted as a set of frames that will be rendered instead of the
+        full rop nodes frame range.
+
+        Only `instance.data["instance_node"]` is required.
+        """
+        # Log the start of the render
+        rop_node = hou.node(instance.data["instance_node"])
+        self.log.debug(f"Rendering {rop_node.path()}")
+
+        frames_to_fix = clique.parse(instance.data.get("frames_to_fix", ""),
+                                     "{ranges}")
+        if len(set(frames_to_fix)) < 2:
+            render_rop(rop_node)
+            return
+
+        # Render only frames to fix
+        for frame_range in frames_to_fix.separate():
+            frame_range = list(frame_range)
+            frame_range.sort()
+            first_frame = int(frame_range[0])
+            last_frame = int(frame_range[-1])
```

Comment:
I just realized that there are `int` conversions, so original `frame_range` might not contain `int`? If yes, then the sort would not work.
```suggestion
            frame_range = [int(frame) for frame in frame_range]
            frame_range.sort()
            first_frame, last_frame = frame_range[0], frame_range[-1]
```

Reply:
I don't think we need to sort since `frame_range` collection is ordered and converting to list won't affect that, no?

```suggestion
            frame_range = list(frame_range)
            first_frame = int(frame_range[0])
            last_frame = int(frame_range[-1])
```

> so original `frame_range` might not contain `int`?

I don't know tbh what would be the padding used in clique when using `{ranges}` keyword without specifying `padding`

Reply:
> I don't think we need to sort since frame_range collection is ordered and converting

If that's the case, then your suggestion is correct.